src/lib.rs

```diff
@@ -95,6 +98,13 @@ impl<A: Array> Drop for ArrayVec<A> {
     }
 }
 
+macro_rules! panic_oob {
+    ($method_name:expr, $index:expr, $len:expr) => {
+        panic!(concat!("ArrayVec::", $method_name, ": index {} is out of bounds in vector of length {}"),
+               $index, $len)
+    }
+}
```

Comment:
Right. The approach is to more or less go for correct with this PR and safe fast for later. I'm not sure we can find a difference, but it would be nice to at least put the panic formatting code in a place that is independent of the type parameters.